meta-mbl-distro/conf/distro/include/mbl-distro.inc

```diff
@@ -31,6 +31,11 @@ BUILDHISTORY_COMMIT = "1"
 
 IMAGE_FSTYPES = "ext4 wic.gz wic.bmap tar.xz"
 
+# Create minimal rootfs file system images
+IMAGE_ROOTFS_SIZE = "0"
+IMAGE_OVERHEAD_FACTOR = "1"
```

Comment:
By reading the [Yocto manual](https://www.yoctoproject.org/docs/current/mega-manual/mega-manual.html#var-IMAGE_OVERHEAD_FACTOR) we need to have some extra space for the post install scripts processing. I'd rather use at least 1.1 factor.

Reply:
But at the same time the manual is not clear if this overhead factor is only applied when the whole rootfs processing was finished.

Reply:
Looking at the `image.bbclass` it seems that the `IMAGE_OVERHEAD_FACTOR` is taking into account after the rootfs has being generated. So, I think that in this case, it is safe to use the factor 1.

Reply:
Thanks for checking; I'm still a bit unsure about this.

IIUC the rootfs size we'll get given the `IMAGE_ROOTFS_SIZE` and `IMAGE_OVERHEAD_FACTOR` that I set is still determined using the output of `du` for the sysroot, which is a bit dodgy, especially if we don't add a bit extra to make up for any differences between what `du` thinks and what is actually required.

Unfortunately, I don't think there's a way to create an ext4 image that's exactly the right size for a bunch of files - you've got to specify the fs size when creating the image before copying the data into it.

I'm not too worried about this because:
* If the rootfs size is too small we should get an error at build time. If there's a problem at runtime then it means were writing to the rootfs at runtime, which we shouldn't be doing.
* Hopefully once we switch to squashfs BitBake will provide a mechanism to do exactly what we want without worrying about these numbers.

Reply:
Another thing that came up in my mind. Should we have, for some reason, an image size that is multiple of flash erase size?

Reply:
I don't *think* that would buy us anything.

Something that might be useful is tuning the sizes that the file system uses for various types of "chunk" based on properties of the flash storage. I don't think it's worth bothering with that before we switch to squashfs though.

Reply:
I think tuning the file system's "chunk" sizes would be much more useful for the r/w partitions TBH - after all, the flash erase size is only a factor when doing writes.